### PR TITLE
add ServersTransport on services

### LINKED ISSUE #4835: Server's transport on services
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

The idea is to be able to specify a different transport configuration dynamically on each service.

**NOTE:** During the refresh of the dynamic config, if the serverTransport doesn't change, we need to use the same `Transport` instance to keep keep-alive connection.

```toml
[services.foobar1.loadbalancer]
serverTransport="custom"

[services.foobar2.loadbalancer]
# as serverTransport is not defined use the ServerTransports named "default"
```

```toml
[ServerTransports]

[ServerTransports.custom]
  InsecureSkipVerify = true
  RootCAs = ["foobar", "foobar"]
  MaxIdleConnsPerHost = 42
    [ServerTransports.custom.ForwardingTimeouts]
    DialTimeout = 42
    ResponseHeaderTimeout = 42

[ServerTransports.default]
  InsecureSkipVerify = true
  RootCAs = ["foobar", "foobar"]
  MaxIdleConnsPerHost = 42
    [ServerTransports.default.ForwardingTimeouts]
    DialTimeout = 42
    ResponseHeaderTimeout = 42
```

### PR DESCRIPTION
**Pull request by juliens**
### What does this PR do?

This PR adds the support of serversTransport per services.

### Motivation
Fixes #4835 

### More

- [x] Added/updated tests
- [x] Added/updated documentation

### CODE REVIEW

**Review by rtribotte — APPROVED** [review]
LGTM 👍

File: pkg/provider/kubernetes/crd/fixtures/with_servers_transport.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+kind: Secret
+metadata:
+  name: rootCas1
+  namespace: foo
+
+data:
+  tls.ca: VEVTVFJPT1RDQVM=
+
+---
+
```
Comment (kevinpollet) [inline pkg/provider/kubernetes/crd/fixtures/with_servers_transport.yml:11]:
```suggestion
```

File: pkg/provider/kubernetes/crd/traefik/v1alpha1/serverstransport.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+// +k8s:deepcopy-gen=true
+
+// ServersTransportSpec options to configure communication between Traefik and the servers.
+type ServersTransportSpec struct {
+	ServerName          string              `description:"ServerName used to contact the server" json:"serverName,omitempty" toml:"serverName,omitempty" yaml:"serverName,omitempty" export:"true"`
+	InsecureSkipVerify  bool                `description:"Disable SSL certificate verification." json:"insecureSkipVerify,omitempty" toml:"insecureSkipVerify,omitempty" yaml:"insecureSkipVerify,omitempty" export:"true"`
+	RootCAsSecrets      []string            `description:"Add cert file for self-signed certificate." json:"rootCAsSecrets,omitempty" toml:"rootCAsSecrets,omitempty" yaml:"rootCAsSecrets,omitempty"`
+	CertificatesSecrets []string            `description:"Certificates for mTls." json:"certificatesSecrets,omitempty" toml:"certificatesSecrets,omitempty" yaml:"certificatesSecrets,omitempty"`
```
Comment (kevinpollet) [inline pkg/provider/kubernetes/crd/traefik/v1alpha1/serverstransport.go:26]:
```suggestion
	CertificatesSecrets []string            `description:"Certificates for mTLS." json:"certificatesSecrets,omitempty" toml:"certificatesSecrets,omitempty" yaml:"certificatesSecrets,omitempty"`
```

**Review by kevinpollet — COMMENTED** [review]
Just minor suggestions :D

LGTM 👍

**Review by ldez — APPROVED** [review]
LGTM

**Review by jbdoumenjou — APPROVED** [review]
LGTM

**Comment by Lobstrosity** [comment]
Am I reading the comments here and in https://github.com/traefik/traefik/issues/4835 correctly that this change will allow for host-validated HTTPS calls to backend services? (I.e., the equivalent of `curl --resolve $host:$ip:$port https://$host:$port/` instead of the current IP-validated `curl -h 'Host: $host' https://$ip:$port/` behavior?)

**Comment by NoxInmortus** [comment]
Hello, is this released yet ? I dont find it in the releases.

**Comment by jbdoumenjou** [comment]
@NoxInmortus  It will be released in the next minor version. For now, it is only available on experimental version.

**Comment by branchmispredictor** [comment]
Hi there, thanks for this change. Is there an ETA to when this will get released? There have been 2 minor version releases of 2.3 Traefik since this was merged, but I do not see this change in either of them.

**Comment by jbdoumenjou** [comment]
Hi,
Traefik [releases](https://github.com/traefik/traefik/#release-cycle) follow the [semantic versionning](https://semver.org/).
I suppose you are mixing-up minor and patch versions.

**Comment by branchmispredictor** [comment]
> Hi,
> Traefik [releases](https://github.com/traefik/traefik/#release-cycle) follow the [semantic versionning](https://semver.org/).
> I suppose you are mixing-up minor and patch versions.

Yes, you're right of course. There have been 2 patch versions, my apologies there. Will this be released as a 2.3 patch, or will it be slated for a later minor version release?

**Comment by ldez** [comment]
It will be in the next minor version: v2.4

**Comment by flixx** [comment]
Hello,

I'm trying to use traefik on Kubernetes and are looking for a way to disable SSL verification for a single Backend / Pod.

I followed this path until I got here:

https://github.com/traefik/traefik/issues/364 -> https://github.com/traefik/traefik/issues/2367 -> https://github.com/traefik/traefik/issues/3038 -> here

It seems somehow that this PR was addressing this issue, however, I still don't know how to use it. Especially, I am looking for an Ingress Annotation to trust my self-signed pod. Is this possible after this change? And if yes, is it documented?

Thanks a lot!

**Comment by rtribotte** [comment]
Hello @flixx,

You are probably looking for [this option](https://doc.traefik.io/traefik/routing/services/#insecureskipverify).
And the related annotation `traefik.ingress.kubernetes.io/service.serverstransport` (available in v2.5).

**Comment by TheUltimateC0der** [comment]
I am having the same problem as @flixx 

btw ... using nginx ingress with SSL only pods is pretty straight forward compared to this ...

I have set up a completely fresh install of k3s with rancher. Now I want to disable SSL verification in order to be able to connect properly.

traefik version is 2.48, and I cannot upgrade. How can I archive this ?

**Pull request merged**